.github/workflows/build.yml

```diff
@@ -0,0 +1,61 @@
+name: Build and Test
+on:
+  pull_request:
+    branches:
+      - '**'
+    types: [ opened, synchronize, reopened ]
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Install Python dependencies
+        run: |
+          python3 -m pip install --upgrade pip
+          python3 -m pip install -r requirements.txt
+        shell: bash
```

Comment:
You are missing the setup python step which defaults to the `python` executable not `python3` executable. I think this is running thanks to the `ubuntu-latest` having python baked in

https://github.com/actions/setup-python

Reply:
Fixed in commit 61a5cf87ea4f9083ef9ed14eed501c6f90e5a980.